### DISCUSSION TITLE
how to add node ip or hostname attribute to metric

### DISCUSSION THREAD

**Discussion by lazyfighter** [open-telemetry/opentelemetry-specification#3223] [Q&A]
Value: 0
	{"kind": "exporter", "data_type": "metrics", "name": "logging"}
2023-02-17T17:17:57.719+0800	info	MetricsExporter	{"kind": "exporter", "data_type": "metrics", "name": "logging", "#metrics": 24}
2023-02-17T17:17:57.720+0800	info	ResourceMetrics #0
Resource SchemaURL: https://opentelemetry.io/schemas/1.9.0
Resource attributes:
     -> container.id: Str(17480)
     -> host.arch: Str(amd64)
     -> host.name: Str(yz5514)
     -> nodeIp: Str(test)
     -> os.description: Str(Linux 3.10.0-1160.66.1.el7.x86_64)
     -> os.type: Str(linux)
     -> process.command_line: Str(/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b08-1.el7_9.x86_64/jre:bin:java -javaagent:./opentelemetry-javaagent.jar -Dotel.resource.attributes=nodeIp=test -Dotel.service.name=test -Dotel.exporter.otlp.endpoint=http://127.0.0.1:4318)
     -> process.executable.path: Str(/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b08-1.el7_9.x86_64/jre:bin:java)
     -> process.pid: Int(28329)
     -> process.runtime.description: Str(Red Hat, Inc. OpenJDK 64-Bit Server VM 25.362-b08)
     -> process.runtime.name: Str(OpenJDK Runtime Environment)
     -> process.runtime.version: Str(1.8.0_362-b08)
     -> service.name: Str(test)
     -> telemetry.auto.version: Str(1.16.0)
     -> telemetry.sdk.language: Str(java)
     -> telemetry.sdk.name: Str(opentelemetry)
     -> telemetry.sdk.version: Str(1.16.0)
ScopeMetrics #0
ScopeMetrics SchemaURL:
InstrumentationScope io.opentelemetry.runtime-metrics
Metric #0
Descriptor:
     -> Name: process.runtime.jvm.system.cpu.load_1m
     -> Description: Average CPU load of the whole system for the last minute
     -> Unit: 1
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0.000000
Metric #1
Descriptor:
     -> Name: process.runtime.jvm.memory.limit
     -> Description: Measure of max obtainable memory
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 663748608
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 9437184
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 1367343104
NumberDataPoints #3
Data point attributes:
     -> pool: Str(Compressed Class Space)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 1073741824
NumberDataPoints #4
Data point attributes:
     -> pool: Str(Code Cache)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 251658240
Metric #2
Descriptor:
     -> Name: process.runtime.jvm.classes.loaded
     -> Description: Number of classes loaded since JVM start
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 9577
Metric #3
Descriptor:
     -> Name: process.runtime.jvm.cpu.utilization
     -> Description: Recent cpu utilization for the process
     -> Unit: 1
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0.000208
Metric #4
Descriptor:
     -> Name: process.runtime.jvm.classes.unloaded
     -> Description: Number of classes unloaded since JVM start
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0
Metric #5
Descriptor:
     -> Name: process.runtime.jvm.threads.count
     -> Description: Number of executing threads
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 32
Metric #6
Descriptor:
     -> Name: process.runtime.jvm.memory.committed
     -> Description: Measure of memory committed
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 183500800
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 9437184
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 74448896
NumberDataPoints #3
Data point attributes:
     -> pool: Str(Metaspace)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 54919168
NumberDataPoints #4
Data point attributes:
     -> pool: Str(Compressed Class Space)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 7208960
NumberDataPoints #5
Data point attributes:
     -> pool: Str(Code Cache)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 14745600
Metric #7
Descriptor:
     -> Name: process.runtime.jvm.memory.init
     -> Description: Measure of initial memory requested
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 32505856
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 5242880
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 87031808
NumberDataPoints #3
Data point attributes:
     -> pool: Str(Metaspace)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0
NumberDataPoints #4
Data point attributes:
     -> pool: Str(Compressed Class Space)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0
NumberDataPoints #5
Data point attributes:
     -> pool: Str(Code Cache)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 2555904
Metric #8
Descriptor:
     -> Name: process.runtime.jvm.memory.usage
     -> Description: Measure of memory used
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 119007992
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 8662448
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 13409552
NumberDataPoints #3
Data point attributes:
     -> pool: Str(Metaspace)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 51823128
NumberDataPoints #4
Data point attributes:
     -> pool: Str(Compressed Class Space)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 6609544
NumberDataPoints #5
Data point attributes:
     -> pool: Str(Code Cache)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 13455872
Metric #9
Descriptor:
     -> Name: process.runtime.jvm.system.cpu.utilization
     -> Description: Recent cpu utilization for the whole system
     -> Unit: 1
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0.001751
Metric #10
Descriptor:
     -> Name: process.runtime.jvm.classes.current_loaded
     -> Description: Number of classes currently loaded
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 9577
ScopeMetrics #1
ScopeMetrics SchemaURL:
InstrumentationScope io.opentelemetry.runtime-metrics 1.22.1-alpha
Metric #0
Descriptor:
     -> Name: process.runtime.jvm.system.cpu.load_1m
     -> Description: Average CPU load of the whole system for the last minute
     -> Unit: 1
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0.000000
Metric #1
Descriptor:
     -> Name: process.runtime.jvm.classes.loaded
     -> Description: Number of classes loaded since JVM start
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 9577
Metric #2
Descriptor:
     -> Name: process.runtime.jvm.memory.usage_after_last_gc
     -> Description: Measure of memory used after the most recent garbage collection event on this pool
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 8662448
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 13384976
Metric #3
Descriptor:
     -> Name: process.runtime.jvm.memory.limit
     -> Description: Measure of max obtainable memory
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 663748608
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 9437184
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 1367343104
NumberDataPoints #3
Data point attributes:
     -> pool: Str(Compressed Class Space)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 1073741824
NumberDataPoints #4
Data point attributes:
     -> pool: Str(Code Cache)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 251658240
Metric #4
Descriptor:
     -> Name: process.runtime.jvm.cpu.utilization
     -> Description: Recent cpu utilization for the process
     -> Unit: 1
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0.000000
Metric #5
Descriptor:
     -> Name: process.runtime.jvm.classes.unloaded
     -> Description: Number of classes unloaded since JVM start
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0
Metric #6
Descriptor:
     -> Name: process.runtime.jvm.threads.count
     -> Description: Number of executing threads
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> daemon: Bool(true)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 28
NumberDataPoints #1
Data point attributes:
     -> daemon: Bool(false)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 4
Metric #7
Descriptor:
     -> Name: process.runtime.jvm.memory.committed
     -> Description: Measure of memory committed
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 183500800
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 9437184
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 74448896
NumberDataPoints #3
Data point attributes:
     -> pool: Str(Metaspace)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 54919168
NumberDataPoints #4
Data point attributes:
     -> pool: Str(Compressed Class Space)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 7208960
NumberDataPoints #5
Data point attributes:
     -> pool: Str(Code Cache)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 14745600
Metric #8
Descriptor:
     -> Name: process.runtime.jvm.memory.init
     -> Description: Measure of initial memory requested
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 32505856
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 5242880
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 87031808
NumberDataPoints #3
Data point attributes:
     -> pool: Str(Metaspace)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0
NumberDataPoints #4
Data point attributes:
     -> pool: Str(Compressed Class Space)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0
NumberDataPoints #5
Data point attributes:
     -> pool: Str(Code Cache)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 2555904
Metric #9
Descriptor:
     -> Name: process.runtime.jvm.memory.usage
     -> Description: Measure of memory used
     -> Unit: By
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> pool: Str(PS Eden Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 119007992
NumberDataPoints #1
Data point attributes:
     -> pool: Str(PS Survivor Space)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 8662448
NumberDataPoints #2
Data point attributes:
     -> pool: Str(PS Old Gen)
     -> type: Str(heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 13409552
NumberDataPoints #3
Data point attributes:
     -> pool: Str(Metaspace)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 51823128
NumberDataPoints #4
Data point attributes:
     -> pool: Str(Compressed Class Space)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 6609544
NumberDataPoints #5
Data point attributes:
     -> pool: Str(Code Cache)
     -> type: Str(non_heap)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 13455872
Metric #10
Descriptor:
     -> Name: process.runtime.jvm.system.cpu.utilization
     -> Description: Recent cpu utilization for the whole system
     -> Unit: 1
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0.000000
Metric #11
Descriptor:
     -> Name: process.runtime.jvm.classes.current_loaded
     -> Description: Number of classes currently loaded
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 9577
ScopeMetrics #2
ScopeMetrics SchemaURL:
InstrumentationScope io.opentelemetry.sdk.trace
Metric #0
Descriptor:
     -> Name: queueSize
     -> Description: The number of spans queued
     -> Unit: 1
     -> DataType: Gauge
NumberDataPoints #0
Data point attributes:
     -> spanProcessorType: Str(BatchSpanProcessor)
StartTimestamp: 2023-02-17 09:05:57.633 +0000 UTC
Timestamp: 2023-02-17 09:17:57.638 +0000 UTC
Value: 0
	{"kind": "exporter", "data_type": "metrics", "name": "logging"}

**Comment by lazyfighter** [comment]
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: localhost:4318

processors:
  batch:
  resourcedetection:
    detectors: ["env", "system"]
    system:
      hostname_sources: ["os"]
  attributes:
    actions:
      - key: hostname
        value: hostname
        action: insert
      - key: test
        from_attribute: nodeIp
        action: upsert
exporters:
  logging:
    loglevel: debug
  prometheusremotewrite:
    endpoint: "http://10.1.111.68:8428/api/v1/write"
extensions:
  health_check:
  pprof:
  zpages:

service:
  extensions: [health_check, pprof, zpages]
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging]
    metrics:
      receivers: [otlp]
      processors: [resourcedetection,  batch]
      exporters: [prometheusremotewrite,logging]
    logs:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging]

**Comment by lazyfighter** [comment]
processors:
  batch:
  resourcedetection:
    detectors: ["env", "system"]
  attributes:
    actions:
      - key: hostname
        from_attribute: nodeIp
        action: insert
      - key: test
        from_attribute: host.name
        action: insert

**Reply by lazyfighter** [reply]
is still not work

**Comment by rogerogers** [comment]
Can you provide a formatted config? with correct indent

**Reply by lazyfighter** [reply]
receivers:
&nbsp; &nbsp; otlp:
&nbsp; &nbsp; &nbsp; &nbsp;  protocols:
&nbsp; &nbsp; &nbsp; &nbsp;&nbsp; &nbsp;  grpc:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; endpoint: localhost:4318

processors:
&nbsp; &nbsp; batch:
&nbsp; &nbsp; resourcedetection:
&nbsp; &nbsp; &nbsp; &nbsp; detectors: ["env", "system"]
&nbsp; &nbsp; attributes:
&nbsp; &nbsp; &nbsp; &nbsp; actions:
&nbsp; &nbsp; &nbsp; &nbsp; - key: hostname
&nbsp; &nbsp; &nbsp; &nbsp; from_attribute: nodeIp
&nbsp; &nbsp; &nbsp; &nbsp; action: insert
&nbsp; &nbsp; &nbsp; &nbsp;  - key: test
&nbsp; &nbsp; &nbsp; &nbsp; from_attribute: host.name
&nbsp; &nbsp; &nbsp; &nbsp; action: insert
exporters:
&nbsp; &nbsp;  logging:
&nbsp; &nbsp; &nbsp; &nbsp;  loglevel: debug
&nbsp; &nbsp; prometheusremotewrite:
&nbsp; &nbsp; &nbsp; &nbsp; endpoint: "http://10.1.111.68:8428/api/v1/write"
extensions:
&nbsp; &nbsp; health_check:
&nbsp; &nbsp; pprof:
&nbsp; &nbsp; zpages:

service:
&nbsp; &nbsp; extensions: [health_check, pprof, zpages]
&nbsp; &nbsp; pipelines:
 &nbsp; &nbsp; &nbsp; &nbsp; traces:
 &nbsp; &nbsp; &nbsp; &nbsp; receivers: [otlp]
 &nbsp; &nbsp; &nbsp; &nbsp; processors: [batch]
 &nbsp; &nbsp; &nbsp; &nbsp; exporters: [logging]
  &nbsp; &nbsp; metrics:
 &nbsp; &nbsp; &nbsp; &nbsp; receivers: [otlp]
&nbsp; &nbsp; &nbsp; &nbsp; processors: [resourcedetection, attributes, batch]
&nbsp; &nbsp; &nbsp; &nbsp; exporters: [prometheusremotewrite,logging]
&nbsp; &nbsp; logs:
&nbsp; &nbsp; &nbsp; &nbsp; receivers: [otlp]
&nbsp; &nbsp; &nbsp; &nbsp; processors: [batch]
&nbsp; &nbsp; &nbsp; &nbsp; exporters: [logging]

**Reply by lazyfighter** [reply]
java -jar -javaagent:./opentelemetry-javaagent.jar -Dotel.resource.attributes=nodeIp=test -Dotel.service.name=test -Dotel.exporter.otlp.endpoint=http://127.0.0.1:4318 opentelemetry-test-0.0.1-SNAPSHOT.jar

**Reply by lazyfighter** [reply]
@rogerogers

**Comment by rogerogers** [comment]
Seems labels of metrics are limited [metrics](https://opentelemetry.io/docs/reference/specification/metrics/semantic_conventions/http-metrics/#metric-httpserverduration)

**Reply by lazyfighter** [reply]
it doesn`t http metrics, i want to add ip attribute with jvm metircs

**Reply by rogerogers** [reply]
[runtime](https://opentelemetry.io/docs/reference/specification/metrics/semantic_conventions/runtime-environment-metrics/#jvm-metrics) labels are predefined

**Reply by lazyfighter** [reply]
yeah, i konw, runtime label its only contains jvm label, it`s doen`t contains machine atrribute， i want to monitor jvm at machine one by one

**Reply by rogerogers** [reply]
Using the Prometheus exporter and using Prometheus relabel config is a choice.

**Reply by lazyfighter** [reply]
i want to use opentelemetry , look at my jar start options and collector config, i dont know why cant get resource attribute

**Reply by rogerogers** [reply]
```yaml
processors:
  resource:
    attributes:
      - key: you_r
        from_attribute: instance
        action: insert
      - key: you_ip
        from_attribute: node.ip
        action: insert
```
You should use resourceprocessor get resource values. But this seems only affect`target_info` metrics.

**Reply by lazyfighter** [reply]
it not work

**Reply by rogerogers** [reply]
> it not work

Search`target_info` in your metrics list

**Reply by rogerogers** [reply]
> i want to use opentelemetry , look at my jar start options and collector config, i dont know why cant get resource attribute

```yaml
exporters:
  prometheus:
    endpoint: "0.0.0.0:1234"
    namespace: default
    send_timestamps: true
    metric_expiration: 180m
    resource_to_telemetry_conversion:
      enabled: true
```
```bash
curl localhost:1234/metrics
```
Using Prometheus exporter will expose all attributes.

**Reply by lazyfighter** [reply]
i will try it , thank you~
